### ISSUE TITLE
TextField's caret shifts to the left after initial rendering

### ISSUE THREAD

**Issue by tvolkert** [flutter/flutter#58925]
## Steps to reproduce

Run the following app, and switch focus from the first text field to the second, then back again (as many times as you like):

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(ShiftingCaret());
}

class ShiftingCaret extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'TextField',
      home: Scaffold(
        body: Padding(
          padding: EdgeInsets.all(10),
          child: Row(
            children: [
              SizedBox(width: 100, child: TextField()),
              SizedBox(width: 100, child: TextField()),
            ],
          ),
        ),
      ),
    );
  }
}
```

## Expected behavior

As the caret appears and begins blinking in the focused text field, you expect the caret's offset (position) within the text field to remain fixed until you start typing.

## Actual behavior

The caret starts in one position, then quickly shifts to the left as it begins blinking.  It's a jarring experience.

This bug happens both in mobile and in desktop/web.  However, in a desktop or web context, where the user can use the TAB keyboard key to quickly navigate from one text field to the next, it's especially pronounced and unsettling.

![ezgif-6-df58992f8781](https://user-images.githubusercontent.com/15253456/83974400-ec8bb000-a8a1-11ea-9747-704c3ab9be2d.gif)

![ezgif-6-311bf620e3ab](https://user-images.githubusercontent.com/15253456/83974468-573ceb80-a8a2-11ea-8b0f-dc0947adc809.gif)

Other facets of this bug:
1. The caret only behaves this way upon the text field gaining focus when the text filed is empty.  Once the text field has content, this behavior does not manifest.
1. Once the text field has content, if the user uses the keyboard to move the caret back to position zero, the shift can be seen again, and it now shifts not only the caret, _but all the text as well_.

![ezgif-6-bd5e98abf723](https://user-images.githubusercontent.com/15253456/83974514-a256fe80-a8a2-11ea-91fe-21acdaedf0c4.gif)

This can be experienced at https://dartpad.dev/b5a2a62bfd7c01cc6205ad1ea4785baa

**Comment by HansMuller** [comment]
CC @justinmc

**Comment by justinmc** [comment]
@tvolkert Do you know what sha you were on when you saw this?  I'm strangely unable to reproduce this on Chrome, Android emulator, or iOS emulator.

![cursor_jump](https://user-images.githubusercontent.com/389558/84060209-ae10f680-a970-11ea-87d6-ec0ef61b5069.gif)

**Comment by tvolkert** [comment]
@justinmc I'm synced to eacd1c8de487a22d2158ee723283c67c45433d72.  I'll try to sync to ToT and see if this goes away.

**Comment by justinmc** [comment]
I can reproduce at eacd1c8de, let me know what you see and hopefully this is good to close.

**Comment by tvolkert** [comment]
Indeed, I can't reproduce it at ToT.  I ran a bisect, and it looks like it was fixed last week by bbb95e5787db0780d5fe1abbe023fe771137d325.

FYI @LongCatIsLooong
